plugins/inputs/aos/restapi/aosrestapi.go

```diff
@@ -253,7 +253,9 @@ func (api *AosServerApi) StartStreaming(streamingType string, address string, po
 
   defer resp.Body.Close()
 
-  if resp.StatusCode != 201 {
+  if resp.StatusCode == 409 {
```

Comment:
This is not foolproof as server may start using this code to mean different things while still being compliant with loosely defined specification around this status code in HTTP. So ideally we should list the endpoints and skip adding them if they are there..

BTW do we know when this occurs? Our agent here defines `Stop` which cleans all registered endpoints so in happy path there should be no lingering streaming config endpoints..

Anyways we may be ok with 409 for now as I checked AOS implementation and it's using this code only to mean already existing endpoint today.

Reply:
Thanks.
This is ran anytime AOSOM is started. So yes, it will not have issues the first run on a fresh build but will anytime the service is restarted.